digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/strategy/DefaultSectionCalculatorStrategy.scala

```diff
@@ -118,36 +144,35 @@ class DefaultSectionCalculatorStrategy extends RoadAddressSectionCalculatorStrat
                                 calibrationPoints: Seq[UserDefinedCalibrationPoint]): Point = {
+
     def calibrationPointToPoint(calibrationPoint: UserDefinedCalibrationPoint): Option[Point] = {
       val link = oldLinks.find(_.id == calibrationPoint.projectLinkId).orElse(newLinks.find(_.id == calibrationPoint.projectLinkId))
       link.flatMap(pl => GeometryUtils.calculatePointFromLinearReference(pl.geometry, calibrationPoint.segmentMValue))
     }
 
-    //First it will return the subtraction between endPoint - startPoint of all the points variable
-    //Then it will sum all the results of the previous operation as a vector (v1 is the accumulated result, v2 is the next item to sum
-    def calculateDirectionVector(projectLinkSeq: Seq[ProjectLink]): Vector3d = {
-      projectLinkSeq.map(pl => (pl.startingPoint, pl.endPoint)).map(p => p._2 - p._1).fold(Vector3d(0, 0, 0)) { case (v1, v2) => v1 + v2 }.normalize2D()
-    }
-
     // Pick the one with calibration point set to zero: or any old link with lowest address: or new links by direction
     calibrationPoints.find(_.addressMValue == 0).flatMap(calibrationPointToPoint).getOrElse(
       oldLinks.filter(_.status == LinkStatus.UnChanged).sortBy(_.startAddrMValue).headOption.map(_.startingPoint).getOrElse {
         val remainLinks = oldLinks ++ newLinks
         if (remainLinks.isEmpty)
           throw new InvalidAddressDataException("Missing right track starting project links")
-        //Grab all the endpoints of the links
-        val points = remainLinks.map(pl => (pl.startingPoint, pl.endPoint))
-        //TODO: ORIGINAL Direction calculation
-        val direction = points.map(p => p._2 - p._1).fold(Vector3d(0, 0, 0)) { case (v1, v2) => v1 + v2 }.normalize2D()
-        /*TODO: VIITE-1601 direction calculation
-        val direction = remainLinks.exists(_.sideCode != SideCode.Unknown) match {
-          case true =>
-            //We use the points that already have a side code defined as the basis to find the direction vector
-            calculateDirectionVector(remainLinks.filter(_.sideCode != SideCode.Unknown))
-          case _ =>
-            calculateDirectionVector(remainLinks)
+        // Grab all the endpoints of the links
+        val direction = if (remainLinks.exists(_.sideCode != SideCode.Unknown)) {
+          remainLinks.filter(_.sideCode != SideCode.Unknown).map(p => p.endPoint - p.startingPoint).fold(Vector3d(0, 0, 0)) { case (v1, v2) => v1 + v2 }.normalize2D()
+        } else {
+          val sampleLink = remainLinks.head.geometry.head
+          if (remainLinks.map(p => GeometryUtils.geometryEndpoints(p.geometry)).forall(geo => geo._1.y == sampleLink.y && geo._2.y == sampleLink.y)) {
+            // Very rare case
+            // Start and end point of the geometry Y Values are equal on the tracks check the X Value
+            Vector3d(1.0, 0.0, 0.0)
+          } else {
+            // Use the default vector
+            Vector3d(0.0, 1.0, 0.0)
+          }
```

Comment:
The direction used on the `pl.getEndPoints(direction)` can not be the same if exists at least one project link with all points in the same Y.